Refresh.GameServer/Services/MatchService.cs

```diff
@@ -21,38 +21,56 @@ public partial class MatchService(Logger logger) : EndpointService(logger)
 
     public IRoomAccessor RoomAccessor { get; private set; } = null!; //initialized in Initialize()
     
-    public GameRoom GetOrCreateRoomByPlayer(GameUser player, TokenPlatform platform, TokenGame game, NatType natType)
+    public GameRoom GetOrCreateRoomByPlayer(
+        GameUser player, 
+        TokenPlatform platform, 
+        TokenGame game, 
+        NatType natType, 
+        int? buildVersion,
+        bool passedNoJoinPoint)
     {
         GameRoom? room = this.RoomAccessor.GetRoomByUser(player, platform, game);
```

Comment:
I see this being initialized to `false` a lot depending on a nullable value, which makes me think it should work like this:

```suggestion
        bool? passedNoJoinPoint = null)
    {
        passedNoJointPoint ??= false;
        GameRoom? room = this.RoomAccessor.GetRoomByUser(player, platform, game);
```

And then a consumer can just pass in the `Nullable<bool>` without a `?? false`.